server/bundler/assets-writer.js

```diff
@@ -0,0 +1,100 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+const fs = require( 'fs' ); // eslint-disable-line  import/no-nodejs-modules
+const path = require( 'path' );
+
+function AssetsWriter( options ) {
+	this.options = Object.assign(
+		{
+			path: './build',
+			filename: 'assets.json',
+		},
+		options
+	);
+	this.createOutputStream();
+}
+
+Object.assign( AssetsWriter.prototype, {
+	createOutputStream: function() {
+		this.outputPath = path.join( this.options.path, this.options.filename );
+		this.outputStream = fs.createWriteStream( this.outputPath );
+	},
+	apply: function( compiler ) {
+		const self = this;
+		compiler.plugin( 'after-emit', function( compilation, callback ) {
+			const stats = compilation.getStats().toJson( {
+				hash: true,
+				publicPath: true,
+				assets: true,
+				children: false,
+				chunks: true,
+				chunkModules: false,
+				chunkOrigins: false,
+				entrypoints: true,
+				modules: false,
+				source: false,
+				errorDetails: true,
+				timings: false,
+				reasons: false,
+			} );
+
+			const statsToOutput = {};
+			statsToOutput.publicPath = stats.publicPath;
+			statsToOutput.manifests = {};
+
+			for ( const name in stats.assetsByChunkName ) {
```

Comment:
we can use `_.find` here

Reply:
this is a loop to protect us from a future where we have more than one entrypoint, and therefore, more than one manifest.

Reply:
ok. i think thats looking towards a future that might happen, but might never happen.  I'm personally not that very into supporting potential future use-cases just in case [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

even so, we could do something like:
```js
const manifestChunks =  Object.keys( stats.assetsByChunkName )
  .filter( name => name.startsWith( 'manifest' ) )
  .map( name => [ name, stats.assetsByChunkName[ name ].source() ] );

statsToOutput.manifests = fromPairs( manifestChunks );
```

Reply:
Whereas my favorite solution would be to code for the now, and write something like:
```js
statsToOutput.manifests = { manifest: stats.assetsByChunkName[ 'manifest' ].source() }
```